task/sast-find-unicode-control/0.1/sast-find-unicode-control.yaml

```diff
@@ -0,0 +1,232 @@
+                show_error
+                return 1
+            else
+                echo "Message: Successfully cloned the repository: ${FIND_UNICODE_CONTROL_GIT_URL}" >&2
+            fi
+        
+            # Check usage of find-unicode-control to confirm the tool can work
+            ./cloned/find_unicode_control.py --help >&2
+            status=$?
+            if [ "${status}" -ne 0 ]; then
+                echo "Error: Failed to check usage of find-unicode-control" >&2
+                show_error
+                return 1
+            else
+                echo "Message: Succeed to check usage of find-unicode-control" >&2
+            fi
+        }
+        
+        # Find unicode control
+        unicontrol_scan()
+        {
+            LANG=en_US.utf8 ./cloned/find_unicode_control.py -d -v -p bidi --notests "$SOURCE_CODE_DIR" >raw_sast_find_unicode_control_check_out.txt 2>raw_sast_find_unicode_control_check_out.log
+            status=$?
+            if [ "${status}" -ne 0 ] && [ "$status" -ne 1 ]; then
+                echo "Error: sast-find-unicode-control test failed because of the following issues:"  >&2
+                cat raw_sast_find_unicode_control_check_out.log
+                show_error
+                return 1
+            else
+                echo "Message: Succeed to run find-unicode-control" >&2
+            fi
+        
+            # translate the output format
+            sed -i raw_sast_find_unicode_control_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|'
+            status=$?
+            if [ "${status}" -ne 0 ] && [ "${status}" -ne 1 ]; then
+                echo "Error: failed to translate the unicontrol output format" >&2
+                show_error
+                return 1
+            else
+                echo "Message: Succeed to translate the unicontrol output format\n" >&2
+                SCAN_RESULT="raw_sast_find_unicode_control_check_out.txt"
+            fi
+        }
+        
+        # Filter known false positive
+        filter_kfp()
+        {
+            # We check if the KFP_GIT_URL variable is set to apply the filters or not
+            if [[ -z "${KFP_GIT_URL}" ]]; then
+                echo "KFP_GIT_URL variable not defined. False positives won't be filtered" >&2
+                mv sast_snyk_check_out_all_findings.json filtered_sast_snyk_check_out.json
+                return 0
+            else
+                echo "Filtering false positives in results files using csfilter-kfp..." >&2
+                cmd=（
+                    csfilter-kfp
+                    --verbose 
+                    --kfp-git-url="${KFP_GIT_URL}"
+                ）
+
+                if [[ -z "${PROJECT_NVR}" ]]; then
+                    cmd+=(--project-nvr="${PROJECT_NVR}")
+                if
+
+                if [[ -z "${RECORD_EXCLUDED}" ]]; then
+                    cmd+=(--record-excluded="${RECORD_EXCLUDED}")
```

Comment:
What is the point of writing the file if it is not uploaded?  The file will be removed as soon as the CI pipeline finishes.

Reply:
Updated code, if the excluded file exist, also upload this file